client/app/contexts/gameSetupContext/index.tsx

```diff
@@ -0,0 +1,99 @@
+import React, {
+  createContext,
+  FC,
+  ReactNode,
+  useContext,
+  useEffect,
+  useMemo,
+  useState,
+} from "react";
+import { TChallengeTimeOption } from "../../types";
+import ChallengeService from "/imports/client/service/ChallengeService";
+
+interface IGameSetupContextValue {
+  challengeTimeOptions: TChallengeTimeOption[];
+}
+
+interface IGameSetupContextProps {
+  challengeService: ChallengeService;
+  children: ReactNode;
+}
+
+const GameSetupContext = createContext<IGameSetupContextValue>({
+  challengeTimeOptions: [],
+});
+
+export const useGameSetup = () => useContext(GameSetupContext);
+
+export const GameSetupProvider: FC<IGameSetupContextProps> = ({
+  challengeService,
+  children,
+}) => {
+  const [challengeTimeOptions, setChallengeTimeOptions] = useState<
+    TChallengeTimeOption[]
+  >([]);
+
+  const contextValue = useMemo(
+    () => ({
+      challengeTimeOptions,
+    }),
+    [challengeTimeOptions],
+  );
+
+  useEffect(() => {
+    challengeService.buttonEvents.on("ready", () => {
+      const challengeButtons = challengeService.getButtons() || [];
+
+      const newTimeOptions = challengeButtons.map<TChallengeTimeOption>(
```

Comment:
please, validate the buttons. Every button should be unique with minutes. Also, you forgot about one thing. At first, there is a name for each button (for what?). The second one, do not forget about the opponent's clock.